packages/toolpad-app/src/components/AppEditor/PageEditor/RenderPanel.tsx

```diff
@@ -619,106 +529,334 @@ export default function RenderPanel({ className }: RenderPanelProps) {
         return;
       }
 
-      const slotIndex = findActiveSlotAt(
+      const activeDropNodeId = findNodeAt(
         availableDropTargets,
-        nodesInfo,
-        slots,
+        dropAreaRects,
         cursorPos.x,
         cursorPos.y,
       );
 
-      const activeSlot = slotIndex;
-
       event.preventDefault();
-      if (activeSlot) {
-        api.nodeDragOver(activeSlot);
-      } else {
-        api.nodeDragOver(null);
+
+      const activeDropNode = activeDropNodeId && appDom.getNode(dom, activeDropNodeId);
+
+      let activeDropZone = null;
+      if (activeDropNode) {
+        const activeDropNodeRect = dropAreaRects[activeDropNodeId];
+        const activeNodeDropInfo = nodesInfo[activeDropNodeId];
+
+        const isDraggingOverPage = appDom.isPage(activeDropNode);
+        const isDraggingOverRow = appDom.isElement(activeDropNode) && isPageRow(activeDropNode);
+        const isDraggingOverContainer = activeNodeDropInfo
+          ? hasContainerComponent(activeNodeDropInfo)
+          : false;
+
+        let centerAreaFraction = 0;
+        if (isDraggingOverContainer) {
+          centerAreaFraction = 0.625;
```

Comment:
I believe we can make this binary. no need for `centerAreaFraction`. Either it's a container and we detect overlap with `rectContainsPoint` or it's not and we detect the edge using the quadrants algorithm.
(I'd also name it `getDiagonalQuadrant(rectangle, x, y)` or so to be a bit more descriptive)

Reply:
i see, this answers my previous comment, ok let's try that